### PR TITLE
Upgrade to Storyblocks API Version 2

### PR DESCRIPTION
**Pull request by aeh**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,119 +22,118 @@ const credentials = {
...
+mostly, but not entirely, snake case values returned from Storyblocks).
 
 ### Audioblocks
 
 #### Search Audio
 ```js
 const results = await audioblocks.search(params);
 ```
-For a list of parameters see the [docs](https://developer.storyblocks.com/docs/v1/index.html#/?id=search-audio).
+For a list of parameters see the [docs](https://documentation.storyblocks.com/#e4fbb5b0-0864-4e2a-8281-6c12b50f9a3e).
```
Comment (timlesallen) [inline README.md:56]: The URL seems to work OK without the crazy `#...`.  Is it necessary?

File: README.md [hunk 2]
```diff
@@ -22,119 +22,118 @@ const credentials = {
...
 ```
-For a list of parameters see the [docs](https://developer.storyblocks.com/docs/v1/index.html#/?id=list-collections).
+For a list of parameters see the [docs](https://documentation.storyblocks.com/#1debad1c-2135-4163-8a1c-73e662ce3a2b).
 
 #### Get Collections
 ```js
 const results = await videoblocks.collection(params);
 ```
-For a list of parameters see the [docs](https://developer.storyblocks.com/docs/v1/index.html#/?id=get-collection).
+For a list of parameters see the [docs](https://documentation.storyblocks.com/#f60c7ed5-4c16-4c06-a992-04883c2838ba).
```
Comment (timlesallen) [inline README.md:136]: These hashes in the URL could be brittle, not sure we can do anything about it.

File: src/index.js [hunk 3]
```diff
@@ -5,48 +5,34 @@ const APIS = [
...
+      { name: 'categories', endpoint: e`stock-item/categories` },
+      { name: type, endpoint: e`stock-item/details/${'stock_item_id'}` },
+      { name: 'similar', endpoint: e`stock-item/similar/${'stock_item_id'}` },
+      { name: 'download', endpoint: e`stock-item/download/${'stock_item_id'}` },
+      { name: 'collections', endpoint: e`collections` },
+      { name: 'collection', endpoint: e`collections/${'collection_id'}` }
+    ]);
+  });
+  return services;
+}
```
Comment (timlesallen) [inline src/index.js:36]:
```suggestion
function services (credentials) {
  const services = APIS.map(({ service, type, prefixUrl }) => [
    service,
    new StoryblocksApi(prefixUrl, credentials, [
      { name: 'search', endpoint: e`search` },
      { name: 'categories', endpoint: e`stock-item/categories` },
      { name: type, endpoint: e`stock-item/details/${'stock_item_id'}` },
      { name: 'similar', endpoint: e`stock-item/similar/${'stock_item_id'}` },
      { name: 'download', endpoint: e`stock-item/download/${'stock_item_id'}` },
      { name: 'collections', endpoint: e`collections` },
      { name: 'collection', endpoint: e`collections/${'collection_id'}` }
    ])
  ]));
  return new Map(services); // or Object.fromEntries(services)
}
```

File: src/storyblocks.js [hunk 4]
```diff
@@ -1,37 +1,49 @@
...
+const isPlainObject = require('lodash.isplainobject');
 const mapKeys = require('lodash.mapkeys');
 const snakeCase = require('lodash.snakecase');
 
 const AUTH_EXPIRY_SECONDS = 12 * 60 * 60; // 12 hours
 
-// strip any \u0000 null characters from the response before it is json parsed.
+function createError (code, errorOrErrors) {
+  if (typeof errorOrErrors === 'string') return _createError(code, errorOrErrors);
+  const errors = mapKeys(errorOrErrors, (value, key) => camelCase(key));
```
Comment (timlesallen) [inline src/storyblocks.js]:
```suggestion
  const errors = Object.keys().map(key => camelCase(key));
```
Would this not give the same result (and be a little more vanilla)?

Comment (timlesallen) [inline src/storyblocks.js]:
```suggestion
  const errors = Object.keys().map(key => camelCase(key));
```
Would this not give the same result (and be a little more vanilla)?

Comment (timlesallen) [reply src/storyblocks.js]: (or is there a risk of getting inherited keys I'm not accounting for?)

File: src/storyblocks.js [hunk 5]
```diff
@@ -55,11 +68,24 @@ class StoryblocksApi {
...
+   * @param {object} object
    * @return {object}
    */
-  query (params) {
-    return mapKeys(params, (value, key) => snakeCase(key));
+  toCamelCase (object) {
+    return mapKeys(object, (value, key) => {
+      if (/^_\d+p$/.test(key)) return key.slice(1);
+      return camelCase(key);
+    });
```
Comment (timlesallen) [inline src/storyblocks.js]: I'd make a comment about any exceptions catered for here.

File: src/storyblocks.js [hunk 6]
```diff
@@ -55,11 +68,24 @@ class StoryblocksApi {
    * in JS we need to convert them back to snake case before making our
    * requests.
    *
-   * @param {object} params
+   * @param {object} object
+   * @return {object}
+   */
+  toSnakeCase (object) {
+    return mapKeys(object, (value, key) => snakeCase(key));
+  }
```
Comment (timlesallen) [inline src/storyblocks.js]: Does having this as its own function really add much value?

Comment (timlesallen) [reply src/storyblocks.js]: (I mean, we can do a map over the keys inline?)

Comment (timlesallen) [reply src/storyblocks.js]: Also, is there really any point having `toSnakeCase()` and `toCamelCase()` as function on the class?  They don't use `this` at all, they're pure functions.

File: test/videoblocks.js [hunk 7]
```diff
@@ -0,0 +1,140 @@
+const { expect } = require('chai');
+const { getCredentials, nockBack } = require('./util');
+const credentials = getCredentials();
+const { videoblocks } = require('..')(credentials);
+
+describe('service: videoblocks', function () {
+  this.timeout(5000);
```
Comment (timlesallen) [inline test/videoblocks.js:7]: Is this so that when we are running in fixture generation mode, then you we don't time out?

File: test/util.js [hunk 8]
```diff
@@ -0,0 +1,108 @@
+// Load local .env file config if it exists
```
Comment (timlesallen) [inline test/util.js:1]: I think we need a general 'how to record new fixtures/how the whole nockback thing works' comment bit somewhere.  Maybe here?  Maybe in the README?

File: src/storyblocks.js (hunk 4 above)
Comment (aeh) [reply src/storyblocks.js]: @timlesallen not really...
```js
const o = { test_value: 42 };
mapKeys(o, (value, key) => camelCase(key)); // = { testValue: 42 }
Object.keys().map(key => camelCase(key)); // = ['testValue']
```

Comment (aeh) [reply src/storyblocks.js]: @timlesallen https://lodash.com/docs/4.17.15#mapKeys

File: README.md (hunk 2 above)
Comment (aeh) [reply README.md:136]: @timlesallen true but we can't do much about that. At least the underlying page should be relatively stable I hope and even if you aren't taken directly to the correct section you should be able to find it.

File: src/index.js (hunk 3 above)
Comment (aeh) [reply src/index.js:36]: @timlesallen not sure either of those options is actually any easier to read?... and `Map` isn't the right thing in this case really...
```js
audioblocks.get('search')(...args);
```

File: test/videoblocks.js (hunk 7 above)
Comment (aeh) [reply test/videoblocks.js:7]: @timlesallen yes

**Pull request merged**